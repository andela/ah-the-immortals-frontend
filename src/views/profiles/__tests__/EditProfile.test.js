import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ProfileContainer from '../EditProfile';
import ProfileForm from '../../../components/profiles/EditProfile';

describe('Test edit profile container', () => {
  let testStore = configureMockStore([thunk]);
  let store = testStore({Profile: {profile: {}, errors: {} }});

  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <ProfileContainer />
      </BrowserRouter>
    </Provider>
  );

  it('test edit profile container snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('test handle change in form', () => {
    const form = wrapper.find(ProfileForm);
    const userInput = form.find('input[name="first_name"]');
    userInput.simulate('change', {
      preventDefault: jest.fn(),
      target: {
        value: 'abraham',
        name: 'first_name'
      }
    });
    expect(userInput.length).toEqual(1);
  });

  it('test handle submit form', () => {
    const saveProfile = wrapper.findWhere(n => n.type() === 'button' && n.contains('Save'));
    saveProfile.simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(saveProfile.length).toEqual(1);
  });

  it('test handle upload profile picture', () => {
    const uploadPhoto = wrapper.findWhere(n => n.type() === 'button' && n.contains('Upload Photo'));
    uploadPhoto.simulate('click', {
      preventDefault: jest.fn(),
    }, window.cloudinary = {
      openUploadWidget(propsObj, callback) {
        setTimeout(callback(null, { event: 'success', info: { secure_url: 'some-url' } }), window.location.reload = jest.fn(), 0);
      }
    });

    expect(uploadPhoto.length).toEqual(1);
  });
});
