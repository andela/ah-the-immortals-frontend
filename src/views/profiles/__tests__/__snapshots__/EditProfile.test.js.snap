// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test edit profile container test edit profile container snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ProfileContainer)>
        <ProfileContainer
          editProfile={[Function]}
          profile={Object {}}
        >
          <div
            className="container profile-form"
          >
            <ProfileForm
              handleChangeProfile={[Function]}
              handleFileUpload={[Function]}
              handleSubmit={[Function]}
            >
              <div>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="firstname"
                    >
                      First Name
                      <br />
                      <input
                        className="form-control"
                        name="first_name"
                        onChange={[Function]}
                        type="text"
                      />
                    </label>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="lastname"
                    >
                      Last Name
                      <br />
                      <input
                        className="form-control"
                        name="last_name"
                        onChange={[Function]}
                        type="text"
                      />
                    </label>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="bio"
                    >
                      Bio
                      <br />
                      <textarea
                        className="form-control"
                        name="bio"
                        onChange={[Function]}
                      />
                    </label>
                  </div>
                  <div
                    className="btn-group mr-2"
                    role="group"
                  >
                    <button
                      className="btn btn-primary"
                      id="save"
                      type="submit"
                    >
                      Save
                    </button>
                  </div>
                  <div
                    className="btn-group"
                    role="group"
                  >
                    <button
                      className="btn btn-primary"
                      onClick={[Function]}
                      type="submit"
                    >
                      Upload Photo
                    </button>
                  </div>
                </form>
              </div>
            </ProfileForm>
          </div>
        </ProfileContainer>
      </Connect(ProfileContainer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
